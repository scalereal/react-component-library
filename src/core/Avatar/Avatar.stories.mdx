import Avatar, { DisplayMap, SizeMap } from '.';
import { Preview, Story, Meta, Props, ArgsTable, Canvas } from '@storybook/addon-docs/blocks';
import { withKnobs, color, boolean, select } from '@storybook/addon-knobs';

<Meta
    title="Components/Core/Avatar"
    component={Avatar}
    argTypes={{
        size: {
            control: {
                type: 'select',
                options: Object.values(SizeMap)
            },
            description: 'Avatar Sizes `XXS` | `XS` | `S` | `M` | `L` | `XL`'
        },
        res: {
            control: { type: 'text' },
            description: 'Avatar resource type `string` provide url or name'
        },
        bgColor: {
            control: 'color',
            description: '`string`'
        },
        textColor: {
            control: 'color',
            description: '`string`'
        },
        hoverBgColor: {
            control: 'color',
            description: '`string`'
        },
        hoverTextColor: {
            control: 'color',
            description: '`string`'
        },
        hoverShadowColor: {
            control: 'color',
            description: '`string`'
        },
        hoverTextShadowColor: {
            control: 'color',
            description: '`string`'
        },
        hoverEffect: {
            control: { type: 'boolean' },
            description: '`boolean`'
        },
        disabled: {
            control: { type: 'boolean' },
            description: '`boolean`'
        },
        display: {
            control: {
                type: 'select',
                options: Object.values(DisplayMap)
            },
            description: 'type `string` value `inline-block` | `block` '
        },
        ref: {
            table: {
                disable: true
            }
        },
        theme: {
            table: {
                disable: true
            }
        }
    }}
/>

# Avatar

Displays an image or text that represents a user or organization

export const Template = args => (
    <div>
        <Avatar
            display="inline-block"
            res="https://media.istockphoto.com/photos/child-hands-formig-heart-shape-picture-id951945718?k=6&m=951945718&s=612x612&w=0&h=ih-N7RytxrTfhDyvyTQCA5q5xKoJToKSYgdsJ_mHrv0="
            {...args}
        />
        <Avatar display="inline-block" res="pradip bhusnar" {...args} />
    </div>
);

<Preview>
    <Story name="Avatar" args={{ size: 'S' }}>
        {Template.bind({})}
    </Story>
</Preview>

### Color Variations

You can either change the theme or pass the colors as props.

<Preview withToolbar>
    <div>
        <Avatar
            display="inline-block"
            size="S"
            res="Pradip Bhusnar"
            hoverEffect={true}
            bgColor="green"
            textColor="white"
            hoverBgColor="blue"
        />
        <Avatar
            display="inline-block"
            size="S"
            res="Pradip Bhusnar"
            hoverEffect={true}
            bgColor="red"
            textColor="white"
            hoverBgColor="green"
        />
    </div>
</Preview>

### Sizes

There are Six sizs available `XXS` | `XS` | `S` | `M` | `L` | `XL`.

<Preview withToolbar>
    <div>
        <Avatar display="inline-block" size="XXS" res="Pradip Bhusnar" hoverEffect={true} />
        <Avatar display="inline-block" size="XS" res="Pradip Bhusnar" hoverEffect={true} />
        <Avatar display="inline-block" size="S" res="Pradip Bhusnar" hoverEffect={true} />
        <Avatar display="inline-block" size="M" res="Pradip Bhusnar" hoverEffect={true} />
        <Avatar display="inline-block" size="L" res="Pradip Bhusnar" hoverEffect={true} />
        <Avatar display="inline-block" size="XL" res="Pradip Bhusnar" hoverEffect={true} />
    </div>
</Preview>

### Props

<Props story="Avatar" />
