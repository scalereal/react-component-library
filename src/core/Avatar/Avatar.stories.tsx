import { Meta, Story } from '@storybook/react/types-6-0';
import React from 'react';
import Avatar from '.';
import { Props } from './types';

const SizeMap: Props['size'][] = ['ES', 'S', 'M', 'L'];

export default {
    title: 'Components/Core/Avatar',
    component: Avatar,
    argTypes : {
        size : {
            control: {
                type: 'select',
                options: Object.values(SizeMap),
            },
        },
        bgColor : {
            control:'color'
        },
        textColor: {
            control:'color'
        },
        hoverBgColor: {
            control:'color'
        },
        hoverTextColor: {
            control:'color'
        },
        hoverShadowColor: {
            control:'color'
        },
        hoverTextShadowColor: {
            control:'color'
        },
        hoverEffect:{ 
            control: { type: 'boolean'}  
        },
        disabled:{
            control:{type:'boolean'}
        }
    }
} as Meta;

const Template: Story<Props> = ({...args }) => {
    // retrieves the appropriate icon passes it as a component prop
    return <Avatar {...args} />;
};

export const Default = Template.bind({});
Default.args = {
    res: 'https://media.istockphoto.com/photos/child-hands-formig-heart-shape-picture-id951945718?k=6&m=951945718&s=612x612&w=0&h=ih-N7RytxrTfhDyvyTQCA5q5xKoJToKSYgdsJ_mHrv0=',
};

export const ImageAvatar = Template.bind({});
ImageAvatar.args = {
    res:'pradip bhusnar'
};
