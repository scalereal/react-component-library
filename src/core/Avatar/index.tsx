import * as React from 'react';
import { WithStyle } from '../../utils';
import { StyledAvatar } from './styled';
import { AvatarProps } from './types';

export const SizeMap: AvatarProps['size'][] = ['XXS','XS', 'S', 'M', 'L', 'XL'];
export const DisplayMap: AvatarProps['display'][] = ['inline-block','block']

const Avatar:React.FC<AvatarProps> & WithStyle = React.memo(React.forwardRef( (props, ref) => {
    const {size,res='Pradip Bhusnar'} = props;

    const isValidURL = (res:string) => {
        const reso = res.match(/(http(s)?:\/\/.)?(www\.)?[-a-zA-Z0-9@:%._\\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\\+.~#?&//=]*)/g);
        return (reso !== null)
    };
      
    const chars =(name:string)=>{
        const nameSplit = String(name).toUpperCase().split(' ');
        if (nameSplit.length == 1) return nameSplit[0].charAt(0);
        return nameSplit[0].charAt(0) + nameSplit[1].charAt(0);
    }
    

    return (
        <StyledAvatar ref={ref} size={size} {...props}>
            {isValidURL(res)?<img src={res} />:<span>{chars(res)}</span>}
        </StyledAvatar>
    );

}));

Avatar.displayName="Avatar";
Avatar.defaultProps = {
    size:'S',
    hoverEffect: true
};

export default Avatar;
