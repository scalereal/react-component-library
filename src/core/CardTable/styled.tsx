import styled from 'styled-components';
import { cardTableTheme } from '../../theme/core/cardTable';
import { Props } from './types';


export const StyledTh = styled.th<Props>`
   padding: 0.5rem;
`;

export const StyledTr = styled.tr<Props>`
    display: flex;
    justify-content: space-between;
    /* align-content: start; */
    align-items: center;
    /* justify-items: left; */
    margin-bottom: 1rem;
    width: 100%;
    padding: 0.5rem 1rem 0.5rem 1rem;
    background-color: transparentize( white,0.15 );
    border: 1px solid ${pr=>pr.borderColor || cardTableTheme.defaults.borderColor};
    border-radius: 5px;
`;

export const StyledThead = styled.thead<Props>`
    ${StyledTr}{
        background-color:${pr=> pr.headerBgColor || cardTableTheme.defaults.headerBgColor};
    }
`; 

export const StyledTbody = styled.tbody<Props>`
    ${StyledTr}{
        box-shadow: 0px 4px 4px rgba(0, 74, 215, 0.45);
        background-color:${pr=> pr.trBgColor || cardTableTheme.defaults.trBgColor};
        &:nth-child(even){background-color: ${pr=> pr.nthTrBgColor || cardTableTheme.defaults.nthTrBgColor};};
        ${pr => pr.hoverEffect?`&:hover {box-shadow: 0px 4px 19px ${pr.hoverShadowColor || cardTableTheme.defaults.hoverShadowColor};
                                border:2px solid ${pr.hoverBorderColor || cardTableTheme.defaults.hoverBorderColor};
                                background-color:${pr.hoverTrBgColor || cardTableTheme.defaults.hoverTrBgColor};}`:''
                }
    }
`;

export const StyledTd = styled.td`
    display: inline-block;
    padding: 0.5rem;
    text-align: left;
`;

export const StyledTable = styled.table<Props>`
    width:90%;
    margin:auto;
    ${StyledTd},${StyledTh}{
        ${pr=> pr.thData && `width:${100 / pr.thData?.length}%;`}
        text-align: left;
    }
`;