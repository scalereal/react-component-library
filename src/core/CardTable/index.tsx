import React, { FC } from 'react';
import { WithStyle } from '../../utils';
import { StyledTable, StyledTbody, StyledTd, StyledTh, StyledThead, StyledTr } from './styled';
import { Props } from './types';

const CardTable:FC<Props> & WithStyle = React.memo(React.forwardRef( (props, ref) => {
    const {thData,tdData} = props;
    if(thData){
    console.log(100/thData?.length)
    }
    return (
            <StyledTable ref={ref} {...props}>
                <StyledThead {...props}>
                    <StyledTr {...props}>
                        {thData && thData.map(th=><StyledTh key={th.id} {...props}>{th.data}</StyledTh>)}
                    </StyledTr>
                </StyledThead>
                <StyledTbody {...props}>
                    {tdData && tdData.map(tr=><StyledTr key={tr.trId} {...props}> {tr.trData.map(td => <StyledTd key={td.tdId} {...props} >{td.tdData}</StyledTd>)} </StyledTr>)}
                </StyledTbody>
            </StyledTable> 
        )
}));

CardTable.displayName="CardTable";
CardTable.defaultProps = {
    // hoverEffect: false
};

export default CardTable;