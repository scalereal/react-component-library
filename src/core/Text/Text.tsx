import React, { FC } from 'react'
import { WithStyle } from '../../utils'
import * as Styled from './Text.styled'
import { TextProps } from './types'

export const textSizes : TextProps['textSize'][] = ['XS', 'S', 'M', 'L', 'XL']
export const textAlignments : TextProps['textAlign'][] = ['left', 'right', 'center', 'justify', 'justify-all', 'start', 'end', 'match-parent', 'inherit', 'initial', 'unset']
export const textWeights : TextProps['textWeight'][] = ['light', 'regular', 'medium', 'strong', 'extraStrong']

const Text: FC<TextProps> & WithStyle = React.memo(
    React.forwardRef(({ children, textSize, textWeight, textAlign, ...restProps }, ref) => {
        return (
            <Styled.Text  textSize={textSize} textWeight={textWeight} textAlign={textAlign} ref={ref} {...restProps }>
                {children}
            </Styled.Text>
        );
    })
);

Text.defaultProps = {
    lineThrough: false,
    uppercase: false,
    textSize:'S'
};
Text.displayName = 'Text';
Text.Style = Styled.Text;

export default Text;
