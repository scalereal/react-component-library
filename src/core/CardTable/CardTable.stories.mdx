import { CardTable } from '.';
import { Avatar } from '../Avatar';
import { Text } from '../Text/Text';
import { Preview, Story, Meta, Props, ArgsTable, Canvas } from '@storybook/addon-docs/blocks';
import { withKnobs, color, boolean, select } from '@storybook/addon-knobs';

<Meta
    title="Components/Core/CardTable"
    component={CardTable}
    argTypes={{
        hoverEffect: {
            control: { type: 'boolean' }
        },
        borderColor: {
            control: 'color'
        },
        headerBgColor: {
            control: 'color'
        },
        hoverBorderColor: {
            control: 'color'
        },
        hoverShadowColor: {
            control: 'color'
        },
        trBgColor: {
            control: 'color'
        },
        nthTrBgColor: {
            control: 'color'
        },
        hoverTrBgColor: {
            control: 'color'
        }
    }}
/>

# CardTable

Displays a CardTable

export const Template = args => <CardTable {...args} />;

<Preview>
    <Story
        name="CardTable"
        args={{
            thData: [
                { id: 1, data: <Text textWeight="strong">User</Text> },
                { id: 2, data: <Text textWeight="strong">Last Login</Text> },
                { id: 3, data: <Text textWeight="strong">Department</Text> },
                { id: 4, data: <Text textWeight="strong">Status</Text> }
            ],
            tdData: [
                {
                    trId: 1,
                    trData: [
                        {
                            tdId: 1,
                            tdData: (
                                <>
                                    <Avatar size="XXS" display="inline-block" res="Akshay Kumar" />
                                    <Text textSize="XS">Akshay Kumar</Text>
                                </>
                            )
                        },
                        { tdId: 2, tdData: <Text textSize="XS">sep 20 2019</Text> },
                        { tdId: 3, tdData: <Text textSize="XS">Development</Text> },
                        {
                            tdId: 4,
                            tdData: (
                                <Text textSize="XS" textColor="blue">
                                    active
                                </Text>
                            )
                        }
                    ]
                },
                {
                    trId: 2,
                    trData: [
                        {
                            tdId: 1,
                            tdData: (
                                <>
                                    <Avatar size="XXS" display="inline-block" res="Akshay Kumar" />
                                    <Text textSize="XS">Akshay Kumar</Text>
                                </>
                            )
                        },
                        { tdId: 2, tdData: <Text textSize="XS">sep 20 2019</Text> },
                        { tdId: 3, tdData: <Text textSize="XS">Development</Text> },
                        {
                            tdId: 4,
                            tdData: (
                                <Text textSize="XS" textColor="blue">
                                    active
                                </Text>
                            )
                        }
                    ]
                },
                {
                    trId: 3,
                    trData: [
                        {
                            tdId: 1,
                            tdData: (
                                <>
                                    <Avatar size="XXS" display="inline-block" res="Akshay Kumar" />
                                    <Text textSize="XS">Akshay Kumar</Text>
                                </>
                            )
                        },
                        { tdId: 2, tdData: <Text textSize="XS">sep 20 2019</Text> },
                        { tdId: 3, tdData: <Text textSize="XS">Development</Text> },
                        {
                            tdId: 4,
                            tdData: (
                                <Text textSize="XS" textColor="blue">
                                    active
                                </Text>
                            )
                        }
                    ]
                },
                {
                    trId: 4,
                    trData: [
                        {
                            tdId: 1,
                            tdData: (
                                <>
                                    <Avatar size="XXS" display="inline-block" res="Akshay Kumar" />
                                    <Text textSize="XS">Akshay Kumar</Text>
                                </>
                            )
                        },
                        { tdId: 2, tdData: <Text textSize="XS">sep 20 2019</Text> },
                        { tdId: 3, tdData: <Text textSize="XS">Development</Text> },
                        {
                            tdId: 4,
                            tdData: (
                                <Text textSize="XS" textColor="blue">
                                    active
                                </Text>
                            )
                        }
                    ]
                }
            ],
            hoverEffect: true
        }}
    >
        {Template.bind({})}
    </Story>
</Preview>

### Color Variations

You can either change the theme or pass the colors as props.

<Preview withToolbar>
    <div>
        <CardTable
            hoverEffect={true}
            headerBgColor="rgb(32 224 255 / 82%)"
            trBgColor="#ad9d9d91"
            nthTrBgColor="#f5eded"
            thData={[
                { id: 1, data: <Text textWeight="strong">User</Text> },
                { id: 2, data: <Text textWeight="strong">Last Login</Text> },
                { id: 3, data: <Text textWeight="strong">Department</Text> },
                { id: 3, data: <Text textWeight="strong">Status</Text> }
            ]}
            tdData={[
                {
                    trId: 1,
                    trData: [
                        {
                            tdId: 1,
                            tdData: (
                                <>
                                    <Avatar size="XXS" display="inline-block" res="Akshay Kumar" />
                                    <Text textSize="XS">Akshay Kumar</Text>
                                </>
                            )
                        },
                        { tdId: 2, tdData: <Text textSize="XS">sep 20 2019</Text> },
                        { tdId: 3, tdData: <Text textSize="XS">Development</Text> },
                        {
                            tdId: 4,
                            tdData: (
                                <Text textSize="XS" textColor="blue">
                                    active
                                </Text>
                            )
                        }
                    ]
                },
                {
                    trId: 2,
                    trData: [
                        {
                            tdId: 1,
                            tdData: (
                                <>
                                    <Avatar size="XXS" display="inline-block" res="Akshay Kumar" />
                                    <Text textSize="XS">Akshay Kumar</Text>
                                </>
                            )
                        },
                        { tdId: 2, tdData: <Text textSize="XS">sep 20 2019</Text> },
                        { tdId: 3, tdData: <Text textSize="XS">Development</Text> },
                        {
                            tdId: 4,
                            tdData: (
                                <Text textSize="XS" textColor="blue">
                                    active
                                </Text>
                            )
                        }
                    ]
                },
                {
                    trId: 3,
                    trData: [
                        {
                            tdId: 1,
                            tdData: (
                                <>
                                    <Avatar size="XXS" display="inline-block" res="Akshay Kumar" />
                                    <Text textSize="XS">Akshay Kumar</Text>
                                </>
                            )
                        },
                        { tdId: 2, tdData: <Text textSize="XS">sep 20 2019</Text> },
                        { tdId: 3, tdData: <Text textSize="XS">Development</Text> },
                        {
                            tdId: 4,
                            tdData: (
                                <Text textSize="XS" textColor="blue">
                                    active
                                </Text>
                            )
                        }
                    ]
                },
                {
                    trId: 4,
                    trData: [
                        {
                            tdId: 1,
                            tdData: (
                                <>
                                    <Avatar size="XXS" display="inline-block" res="Akshay Kumar" />
                                    <Text textSize="XS">Akshay Kumar</Text>
                                </>
                            )
                        },
                        { tdId: 2, tdData: <Text textSize="XS">sep 20 2019</Text> },
                        { tdId: 3, tdData: <Text textSize="XS">Development</Text> },
                        {
                            tdId: 4,
                            tdData: (
                                <Text textSize="XS" textColor="blue">
                                    active
                                </Text>
                            )
                        }
                    ]
                }
            ]}
        />
    </div>
</Preview>

### Props

<Props story="CardTable" />
