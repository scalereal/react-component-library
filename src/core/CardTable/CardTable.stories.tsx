import { Meta, Story } from '@storybook/react/types-6-0';
import React from 'react';
import CardTable from '.';
import Avatar from '../Avatar';
import Text from '../Text/Text';
import { Props } from './types';


export default {
    title: 'Components/Core/CardTable',
    component: CardTable,
    argTypes : {
        hoverEffect:{ 
            control: { type: 'boolean'}  
        },
        borderColor : {
            control:'color'
        },
        headerBgColor: {
            control:'color'
        },
        hoverBorderColor: {
            control:'color'
        },
        hoverShadowColor:{
            control:'color'
        },
        trBgColor:{
            control:'color'
        },
        nthTrBgColor:{
            control:'color'
        },
        hoverTrBgColor:{
            control:'color'
        }
        
    }
} as Meta;

const Template: Story<Props> = ({...args }) => {
    // retrieves the appropriate icon passes it as a component prop
    return <CardTable {...args} />;
};

export const Default = Template.bind({});
Default.args={
    thData:[{id:1,data:<Text textWeight='strong'>User</Text>},{id:2,data:<Text textWeight='strong'>Last Login</Text>},{id:3,data:<Text textWeight='strong'>Department</Text>},{id:3,data:<Text textWeight='strong'>Status</Text>}],
    tdData:[
        {trId:1,trData:[{tdId:1,tdData:<><Avatar size='XXS' display='inline-block' res="Akshay Kumar"/><Text textSize='XS'>Akshay Kumar</Text></>},
            {tdId:2,tdData:<Text textSize='XS'>sep 20 2019</Text>},{tdId:3,tdData:<Text textSize='XS'>Development</Text>},{tdId:4,tdData:<Text textSize='XS' textColor='blue'>active</Text>}]},
        {trId:2,trData:[{tdId:1,tdData:<><Avatar size='XXS' display='inline-block' res="Akshay Kumar"/><Text textSize='XS'>Akshay Kumar</Text></>},
            {tdId:2,tdData:<Text textSize='XS'>sep 20 2019</Text>},{tdId:3,tdData:<Text textSize='XS'>Development</Text>},{tdId:4,tdData:<Text textSize='XS' textColor='blue'>active</Text>}]},
        {trId:3,trData:[{tdId:1,tdData:<><Avatar size='XXS' display='inline-block' res="Akshay Kumar"/><Text textSize='XS'>Akshay Kumar</Text></>},
            {tdId:2,tdData:<Text textSize='XS'>sep 20 2019</Text>},{tdId:3,tdData:<Text textSize='XS'>Development</Text>},{tdId:4,tdData:<Text textSize='XS' textColor='blue'>active</Text>}]},
        {trId:4,trData:[{tdId:1,tdData:<><Avatar size='XXS' display='inline-block' res="Akshay Kumar"/><Text textSize='XS'>Akshay Kumar</Text></>},
            {tdId:2,tdData:<Text textSize='XS'>sep 20 2019</Text>},{tdId:3,tdData:<Text textSize='XS'>Development</Text>},{tdId:4,tdData:<Text textSize='XS' textColor='blue'>active</Text>}]},
    ],
    hoverEffect:true
}


