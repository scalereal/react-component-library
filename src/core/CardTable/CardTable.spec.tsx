import { render } from '@testing-library/react';
import * as React from 'react';
import CardTable from '.';
import Avatar from '../Avatar';
import Text from '../Text/Text';

const thData=[
    { id: 1, data: <Text textWeight="strong">User</Text> },
    { id: 2, data: <Text textWeight="strong">Last Login</Text> },
    { id: 3, data: <Text textWeight="strong">Department</Text> },
    { id: 4, data: <Text textWeight="strong">Status</Text> }
]
const tdData=[
    {
        trId: 1,
        trData: [
            {
                tdId: 1,
                tdData: (
                    <>
                        <Avatar size="XXS" display="inline-block" res="Akshay Kumar" />
                        <Text textSize="XS">Akshay Kumar</Text>
                    </>
                )
            },
            { tdId: 2, tdData: <Text textSize="XS">sep 20 2019</Text> },
            { tdId: 3, tdData: <Text textSize="XS">Development</Text> },
            {
                tdId: 4,
                tdData: (
                    <Text textSize="XS" textColor="blue">active</Text>
                )
            }
        ]
    },
    {
        trId: 2,
        trData: [
            {
                tdId: 1,
                tdData: (
                    <>
                        <Avatar size="XXS" display="inline-block" res="Akshay Kumar" />
                        <Text textSize="XS">Akshay Kumar</Text>
                    </>
                )
            },
            { tdId: 2, tdData: <Text textSize="XS">sep 20 2019</Text> },
            { tdId: 3, tdData: <Text textSize="XS">Development</Text> },
            {
                tdId: 4,
                tdData: (
                    <Text textSize="XS" textColor="blue">active</Text>
                )
            }
        ]
    },
    {
        trId: 3,
        trData: [
            {
                tdId: 1,
                tdData: (
                    <>
                        <Avatar size="XXS" display="inline-block" res="Akshay Kumar" />
                        <Text textSize="XS">Akshay Kumar</Text>
                    </>
                )
            },
            { tdId: 2, tdData: <Text textSize="XS">sep 20 2019</Text> },
            { tdId: 3, tdData: <Text textSize="XS">Development</Text> },
            {
                tdId: 4,
                tdData: (
                    <Text textSize="XS" textColor="blue">active</Text>
                )
            }
        ]
    },
    {
        trId: 4,
        trData: [
            {
                tdId: 1,
                tdData: (
                    <>
                        <Avatar size="XXS" display="inline-block" res="Akshay Kumar" />
                        <Text textSize="XS">Akshay Kumar</Text>
                    </>
                )
            },
            { tdId: 2, tdData: <Text textSize="XS">sep 20 2019</Text> },
            { tdId: 3, tdData: <Text textSize="XS">Development</Text> },
            {
                tdId: 4,
                tdData: (
                    <Text textSize="XS" textColor="blue">active</Text>
                )
            }
        ]
    }
]

describe('CheckBoxGroup component', () => {
    it('should render with default theme', () => {
        const { container } = render(<CardTable />);
        expect(container).toMatchSnapshot();
    });

    it('should render with all the props given', () => {
        const { container } = render(<CardTable hoverEffect={true}
            headerBgColor="rgb(32 224 255 / 82%)"
            trBgColor="#ad9d9d91"
            nthTrBgColor="#f5eded"
            thData={thData}
            tdData={tdData}/>);
        expect(container).toMatchSnapshot();
    });
});