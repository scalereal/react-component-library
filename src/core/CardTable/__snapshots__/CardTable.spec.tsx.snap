// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBoxGroup component should render with all the props given 1`] = `
<div>
  <table
    class="sc-jSgupP cJDMeT"
  >
    <thead
      class="sc-dlfnbm dygKIA"
    >
      <tr
        class="sc-gsTCUz dhuSOP"
      >
        <th
          class="sc-bdfBwQ eINZyl"
        >
          <span
            class="sc-iBPRYJ kBOVnS"
          >
            User
          </span>
        </th>
        <th
          class="sc-bdfBwQ eINZyl"
        >
          <span
            class="sc-iBPRYJ kBOVnS"
          >
            Last Login
          </span>
        </th>
        <th
          class="sc-bdfBwQ eINZyl"
        >
          <span
            class="sc-iBPRYJ kBOVnS"
          >
            Department
          </span>
        </th>
        <th
          class="sc-bdfBwQ eINZyl"
        >
          <span
            class="sc-iBPRYJ kBOVnS"
          >
            Status
          </span>
        </th>
      </tr>
    </thead>
    <tbody
      class="sc-hKgILt cdTezY"
    >
      <tr
        class="sc-gsTCUz dhuSOP"
      >
         
        <td
          class="sc-eCssSg jqBCNq"
        >
          <div
            class="sc-gKsewC llWDhm"
            display="inline-block"
          >
            <span>
              AK
            </span>
          </div>
          <span
            class="sc-iBPRYJ fHvDlZ"
          >
            Akshay Kumar
          </span>
        </td>
        <td
          class="sc-eCssSg jqBCNq"
        >
          <span
            class="sc-iBPRYJ fHvDlZ"
          >
            sep 20 2019
          </span>
        </td>
        <td
          class="sc-eCssSg jqBCNq"
        >
          <span
            class="sc-iBPRYJ fHvDlZ"
          >
            Development
          </span>
        </td>
        <td
          class="sc-eCssSg jqBCNq"
        >
          <span
            class="sc-iBPRYJ hjvKZl"
          >
            active
          </span>
        </td>
         
      </tr>
      <tr
        class="sc-gsTCUz dhuSOP"
      >
         
        <td
          class="sc-eCssSg jqBCNq"
        >
          <div
            class="sc-gKsewC llWDhm"
            display="inline-block"
          >
            <span>
              AK
            </span>
          </div>
          <span
            class="sc-iBPRYJ fHvDlZ"
          >
            Akshay Kumar
          </span>
        </td>
        <td
          class="sc-eCssSg jqBCNq"
        >
          <span
            class="sc-iBPRYJ fHvDlZ"
          >
            sep 20 2019
          </span>
        </td>
        <td
          class="sc-eCssSg jqBCNq"
        >
          <span
            class="sc-iBPRYJ fHvDlZ"
          >
            Development
          </span>
        </td>
        <td
          class="sc-eCssSg jqBCNq"
        >
          <span
            class="sc-iBPRYJ hjvKZl"
          >
            active
          </span>
        </td>
         
      </tr>
      <tr
        class="sc-gsTCUz dhuSOP"
      >
         
        <td
          class="sc-eCssSg jqBCNq"
        >
          <div
            class="sc-gKsewC llWDhm"
            display="inline-block"
          >
            <span>
              AK
            </span>
          </div>
          <span
            class="sc-iBPRYJ fHvDlZ"
          >
            Akshay Kumar
          </span>
        </td>
        <td
          class="sc-eCssSg jqBCNq"
        >
          <span
            class="sc-iBPRYJ fHvDlZ"
          >
            sep 20 2019
          </span>
        </td>
        <td
          class="sc-eCssSg jqBCNq"
        >
          <span
            class="sc-iBPRYJ fHvDlZ"
          >
            Development
          </span>
        </td>
        <td
          class="sc-eCssSg jqBCNq"
        >
          <span
            class="sc-iBPRYJ hjvKZl"
          >
            active
          </span>
        </td>
         
      </tr>
      <tr
        class="sc-gsTCUz dhuSOP"
      >
         
        <td
          class="sc-eCssSg jqBCNq"
        >
          <div
            class="sc-gKsewC llWDhm"
            display="inline-block"
          >
            <span>
              AK
            </span>
          </div>
          <span
            class="sc-iBPRYJ fHvDlZ"
          >
            Akshay Kumar
          </span>
        </td>
        <td
          class="sc-eCssSg jqBCNq"
        >
          <span
            class="sc-iBPRYJ fHvDlZ"
          >
            sep 20 2019
          </span>
        </td>
        <td
          class="sc-eCssSg jqBCNq"
        >
          <span
            class="sc-iBPRYJ fHvDlZ"
          >
            Development
          </span>
        </td>
        <td
          class="sc-eCssSg jqBCNq"
        >
          <span
            class="sc-iBPRYJ hjvKZl"
          >
            active
          </span>
        </td>
         
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`CheckBoxGroup component should render with default theme 1`] = `
<div>
  <table
    class="sc-jSgupP eXoKSi"
  >
    <thead
      class="sc-dlfnbm hVnoOq"
    >
      <tr
        class="sc-gsTCUz dhuSOP"
      />
    </thead>
    <tbody
      class="sc-hKgILt gTegZF"
    />
  </table>
</div>
`;
